docs/src/modules/java/pages/agents.adoc

```diff
@@ -341,23 +341,42 @@ You may frequently hear people say things like "the LLM can make a call" or "the
 
 The Agent will then in turn execute the tool requested by the LLM, incorporate the tool results into the session context, and then send a new prompt. This will continue in a loop until the LLM no longer indicates it needs to invoke a tool to perform its task.
 
-Tool functions are incredibly easy in Akka. All you need to do is decorate an Agent method with the `@FunctionTool`
-annotation, and Akka will automatically resolve it on your behalf. As an example, you might have an agent that
-recommends outdoor activities, but to do so it needs to know the current weather.
+There are two ways to add function tools to your agent:
 
-Anything "current" is beyond the capabilities of an LLM, so the model response will include information on what tool needs to be invoked.
+1. **Agent-defined function tools** — Define function tools directly within your agent class using the`@FunctionTool`
+annotation. These are automatically registered as available tools.
```

Comment:
what is the scope of this registration? Do we need to be explicit in that the tools are scoped to only be visible within this agent?

Reply:
It's scoped to this agent. 
I think it's clear enough because it's added to the effect returned by this agent.

Reply:
Ok, I see your point. I will make more clear that it's automatically registered as a tool available for this same agent